src/ansys/fluent/core/services/app_utilities.py

```diff
@@ -0,0 +1,252 @@
+        self.service = service
+
+    def get_product_version(self, with_patch: bool = True) -> Any:
+        """Get product version."""
+        request = AppUtilitiesProtoModule.GetProductVersionRequest()
+        response = self.service.get_product_version(request)
+        if with_patch:
+            return f"{response.major}.{response.minor}.{response.patch}"
+        else:
+            return f"{response.major}{response.minor}"
+
+    def get_build_info(self) -> Any:
+        """Get build info."""
+        request = AppUtilitiesProtoModule.GetBuildInfoRequest()
+        response = self.service.get_build_info(request)
+        return MessageToDict(response, preserving_proto_field_name=True)
+
+    def get_controller_process_info(self) -> Any:
+        """Get controller process info."""
+        request = AppUtilitiesProtoModule.GetControllerProcessInfoRequest()
+        response = self.service.get_controller_process_info(request)
+        return {
+            "hostname": response.hostname,
+            "process_id": response.process_id,
+            "working_directory": response.working_directory,
+        }
+
+    def get_solver_process_info(self) -> Any:
+        """Get solver process info."""
+        request = AppUtilitiesProtoModule.GetSolverProcessInfoRequest()
+        response = self.service.get_solver_process_info(request)
+        return {
+            "hostname": response.hostname,
+            "process_id": response.process_id,
+            "working_directory": response.working_directory,
+        }
+
+    def get_app_mode(self) -> Any:
+        """Get app mode.
+
+        Raises
+        ------
+        ValueError
+            If app mode is unknown.
+        """
+        request = AppUtilitiesProtoModule.GetAppModeRequest()
+        response = self.service.get_app_mode(request)
+        match response.app_mode:
+            case AppUtilitiesProtoModule.APP_MODE_UNKNOWN:
+                raise ValueError("Unknown app mode.")
+            case AppUtilitiesProtoModule.APP_MODE_MESHING:
+                return pyfluent.FluentMode.MESHING
+            case AppUtilitiesProtoModule.APP_MODE_SOLVER:
+                return pyfluent.FluentMode.SOLVER
+            case AppUtilitiesProtoModule.APP_MODE_SOLVER_ICING:
+                return pyfluent.FluentMode.SOLVER_ICING
+            case AppUtilitiesProtoModule.APP_MODE_SOLVER_AERO:
+                return pyfluent.FluentMode.SOLVER_AERO
+
+    def start_python_journal(self, journal_name: str | None = None) -> Any:
+        """Start python journal."""
+        request = AppUtilitiesProtoModule.StartPythonJournalRequest()
+        request.journal_name = journal_name
+        response = self.service.start_python_journal(request)
+        return response
+
+    def stop_python_journal(self) -> Any:
+        """Stop python journal."""
+        request = AppUtilitiesProtoModule.StopPythonJournalRequest()
+        response = self.service.stop_python_journal(request)
+        return response
```

Comment:
@hpohekar These two methods return gRPC responses, which should not be the case.